File: Persona.java
public class Persona extends Edad {

    private int horaNacimiento;
    private int diaNacimiento;
    private int mesNacimiento;
    private int añoNacimiento;
    private int horaActual;
    private int diaActual;
    private int mesActual;
    private int añoActual;
    private int restaHora;
    private int restaDia;
    private int restaSemana;
    private int restaTotalDia;
    private int restaMes;
    private int restaTotalMes;
    private int restaAño;

    public Persona(int horaNacimiento, int diaNacimiento, int mesNacimiento, int añoNacimiento, int horaActual, int diaActual, int mesActual, int añoActual) {
        this.horaNacimiento = horaNacimiento;
        this.diaNacimiento = diaNacimiento;
        this.mesNacimiento = mesNacimiento;
        this.añoNacimiento = añoNacimiento;
        this.horaActual = horaActual;
        this.diaActual = diaActual;
        this.mesActual = mesActual;
        this.añoActual = añoActual;
    }

    @Override
    public void calcularEdad() {

        restaHora = this.horaActual - this.horaNacimiento;
        if (restaHora < 16) {
            restaHora =+ 24;
            this.diaActual = -1;
        }else{
            restaHora =+ 0;
        }
        this.resultadoHora = restaHora;

        restaDia = this.diaActual - this.diaNacimiento;
        if (restaDia < 0) {
            restaDia =+ 30;
            this.mesActual = -1;
        }else{
            restaDia =+ 0;
        }
        this.resultadoDia = restaDia;
        int restaSemana = this.resultadoDia/7;
        this.restaTotalDia = restaDia - (restaSemana * 7);
        this.resultadoDia = restaTotalDia;
        this.resultadoSemana = restaSemana;

        restaMes = this.mesActual - this.mesNacimiento;
        if (this.restaMes < 0) {
            this.restaMes =+ 12;
            this.añoActual = -1;
        }else{
            this.restaMes =+ 0;
        }
        this.restaAño = (añoActual - añoNacimiento) * 12;
        this.restaTotalMes = restaMes + restaAño;
        this.resultadoTotalMes = this.restaTotalMes;
    }
}

File: Persona.py
from abc import ABC

import self


class Fecha(ABC):
    _mesEdad = int(0)
    _semanasEdad = int(0)
    _diasEdadTotal = int(0)
    _horasEdad = int(0)

    def __init__(self, mesEdad, semanasEdad, diasEdadTotal, horasEdad):
        self._mesEdad = mesEdad
        self._semanasEdad = semanasEdad
        self._diasEdadTotal = diasEdadTotal
        self._horasEdad = horasEdad

    @staticmethod
    def CalcularEdad(self):
        pass

    def getmesEdad(self):
        return self._mesEdad

    def getsemanasEdad(self):
        return self._semanasEdad

    def getdiasEdadTotal(self):
        return self._diasEdadTotal

    def gethorasEdad(self):
        return self._horasEdad


class Persona(Fecha):

    __anioNacimiento = int(0)
    __anioActual = int(0)
    __mesNacimiento = int(0)
    __mesActual = int(0)
    __diaNacimiento = int(0)
    __diaActual = int(0)
    __horaNacimiento = int(0)
    __horaActual = int(0)

    def __init__(self, mesEdad, semanasEdad, diasEdadTotal, horasEdad, anioNacimiento, anioActual, mesNacimiento,
                 mesActual, diaNacimiento, diaActual, horaNacimiento, horaActual):
        Fecha.__init__(self, mesEdad, semanasEdad, diasEdadTotal, horasEdad)
        self.__anioNacimiento = anioNacimiento
        self.__anioActual = anioActual
        self.__mesNacimiento = mesNacimiento
        self.__mesActual = mesActual
        self.__diaNacimiento = diaNacimiento
        self.__diaActual = diaActual
        self.__horaNacimiento = horaNacimiento
        self.__horaActual = horaActual
        self.CalcularEdad()

    def CalcularEdad(self):

        self._horasEdad = self.__horaActual - self.__horaNacimiento
        if self._horasEdad > 0:
            self._horasEdad += 24
            self.__diaActual -= 1
        else:
            self._horasEdad += 0

        diasEdad = self.__diaActual - self.__diaNacimiento
        if diasEdad > 0:
            diasEdad += 30
            self.__mesActual -= 1
        else:
            diasEdad += 0

        self._semanasEdad = diasEdad // 7

        dias = self._semanasEdad * 7
        self._diasEdadTotal = diasEdad - dias

        meses = self.__anioActual - self.__anioNacimiento
        mesesTotal = meses * 12
        total = self.__mesActual - self.__mesNacimiento
        self._mesEdad = total + mesesTotal
